tests/Fixer/ConstantNotation/NativeConstantInvocationFixerTest.php

```diff
@@ -0,0 +1,247 @@
+
+    /**
+     * @dataProvider provideInvalidConfigurationElementCases
+     *
+     * @param mixed $element
+     */
+    public function testConfigureRejectsInvalidExcludeConfigurationElement($element)
+    {
+        $this->expectException(\PhpCsFixer\ConfigurationException\InvalidConfigurationException::class);
+        $this->expectExceptionMessage(sprintf(
+            'Each element must be a non-empty, trimmed string, got "%s" instead.',
+            \is_object($element) ? \get_class($element) : \gettype($element)
+        ));
+
+        $this->fixer->configure([
+            'exclude' => [
+                $element,
+            ],
+        ]);
+    }
+
+    /**
+     * @dataProvider provideInvalidConfigurationElementCases
+     *
+     * @param mixed $element
+     */
+    public function testConfigureRejectsInvalidIncludeConfigurationElement($element)
+    {
+        $this->expectException(\PhpCsFixer\ConfigurationException\InvalidConfigurationException::class);
+        $this->expectExceptionMessage(sprintf(
+            'Each element must be a non-empty, trimmed string, got "%s" instead.',
+            \is_object($element) ? \get_class($element) : \gettype($element)
+        ));
+
+        $this->fixer->configure([
+            'include' => [
+                $element,
+            ],
+        ]);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideInvalidConfigurationElementCases()
+    {
+        return [
+            'null' => [null],
+            'false' => [false],
+            'true' => [true],
+            'int' => [1],
+            'array' => [[]],
+            'float' => [0.1],
+            'object' => [new \stdClass()],
+            'not-trimmed' => ['  M_PI  '],
+        ];
+    }
+
+    public function testConfigureResetsExclude()
+    {
+        $this->fixer->configure([
+            'exclude' => [
+                'M_PI',
+            ],
+        ]);
+
+        $before = '<?php var_dump(m_pi, M_PI);';
+        $after = '<?php var_dump(m_pi, \\M_PI);';
+
+        $this->doTest($before);
+
+        $this->fixer->configure([]);
+
+        $this->doTest($after, $before);
+    }
+
+    public function testIsRisky()
+    {
+        $fixer = $this->createFixer();
+
+        $this->assertTrue($fixer->isRisky());
+    }
+
+    /**
+     * @dataProvider provideFixWithDefaultConfigurationCases
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithDefaultConfiguration($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideFixWithDefaultConfigurationCases()
+    {
+        return [
+            ['<?php var_dump(NULL, FALSE, TRUE, 1);'],
+            ['<?php echo CUSTOM_DEFINED_CONSTANT_123;'],
+            ['<?php echo m_pi; // Constant are case sensitive'],
+            ['<?php namespace M_PI;'],
+            ['<?php namespace Foo; use M_PI;'],
```

Comment:
please add `echo M_PI;` at the end of this case

Reply:
This would result in:
```diff
-<?php namespace Foo; use M_PI; echo M_PI;
+<?php namespace Foo; use M_PI; echo \M_PI;
```
While with `<?php namespace Foo; use M_PI;` I want to check only trait case.

How would the addition be useful?

Reply:
So please add such case as new case (so we would have both of them).

To have behavior of fixer specified in utest for case when global constant is imported.

Reply:
Done